isobus/src/can_transport_protocol.cpp

```diff
@@ -11,860 +11,1004 @@
+		                                parentPointer);
+	}
+
+	void TransportProtocolManager::TransportProtocolSession::set_state(StateMachineState value)
+	{
+		state = value;
+		timestamp_ms = SystemTiming::get_timestamp_ms();
+	}
+
+	std::uint8_t TransportProtocolManager::TransportProtocolSession::get_cts_response_packet_count() const
+	{
+		return clearToSendPacketCount;
+	}
+	void TransportProtocolManager::TransportProtocolSession::set_cts_response_packet_count(std::uint8_t value)
+	{
+		clearToSendPacketCount = value;
+		timestamp_ms = SystemTiming::get_timestamp_ms();
+	}
+
+	std::uint8_t TransportProtocolManager::TransportProtocolSession::get_cts_response_packet_count_max() const
+	{
+		return clearToSendPacketCountMax;
+	}
+
+	std::uint8_t TransportProtocolManager::TransportProtocolSession::get_last_packet_number() const
+	{
+		return lastPacketNumber;
+	}
+
+	void TransportProtocolManager::TransportProtocolSession::set_last_packet_number(std::uint8_t value)
+	{
+		lastPacketNumber = value;
+		timestamp_ms = SystemTiming::get_timestamp_ms();
 	}
 
-	TransportProtocolManager::~TransportProtocolManager()
+	std::uint8_t TransportProtocolManager::TransportProtocolSession::get_remaining_packets() const
 	{
-		// No need to clean up, as this object is a member of the network manager
-		// so its callbacks will be cleared at destruction time
+		return static_cast<std::uint8_t>(totalNumberOfPackets - lastPacketNumber);
 	}
 
-	void TransportProtocolManager::initialize(CANLibBadge<CANNetworkManager>)
+	std::uint8_t TransportProtocolManager::TransportProtocolSession::get_total_number_of_packets() const
+	{
+		return totalNumberOfPackets;
+	}
+
+	TransportProtocolManager::TransportProtocolManager(CANLibBadge<CANNetworkManager>)
+	{
+	}
+
+	bool TransportProtocolManager::initialize(CANLibBadge<CANNetworkManager>)
 	{
 		if (!initialized)
 		{
 			initialized = true;
-			CANNetworkManager::CANNetwork.add_protocol_parameter_group_number_callback(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolCommand), process_message, this);
-			CANNetworkManager::CANNetwork.add_protocol_parameter_group_number_callback(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolData), process_message, this);
+			CANNetworkManager::CANNetwork.add_protocol_parameter_group_number_callback(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolConnectionManagement), process_message, this);
+			CANNetworkManager::CANNetwork.add_protocol_parameter_group_number_callback(static_cast<std::uint32_t>(CANLibParameterGroupNumber::TransportProtocolDataTransfer), process_message, this);
 		}
+		return initialized;
 	}
 
-	void TransportProtocolManager::process_message(const CANMessage &message)
+	void TransportProtocolManager::process_broadcast_announce_message(const std::shared_ptr<ControlFunction> source,
+	                                                                  std::uint32_t pgn,
```

Comment:
Nit (several places)
```suggestion
	                                                                  std::uint32_t parameterGroupNumber,
```